crates/base/src/js_worker/module_loader.rs

```diff
@@ -143,15 +143,30 @@ impl ModuleLoader for DefaultModuleLoader {
             let module_type = get_module_type(fetched_file.media_type)?;
 
             let code = fetched_file.source;
-            let code = emit_parsed_source(
-                &emit_cache,
-                &parsed_source_cache,
-                &module_specifier,
-                fetched_file.media_type,
-                &code,
-                &emit_options,
-                emit_config_hash,
-            )?;
+            let code = match fetched_file.media_type {
+                MediaType::JavaScript
+                | MediaType::Unknown
+                | MediaType::Cjs
+                | MediaType::Mjs
+                | MediaType::Json => code.into(),
```

Comment:
This should also remove the need to always `emit_parsed_source`. But abstract it from doing so in files that aren't meant to be processed/emitted